qiskit/tools/visualization/_state_visualization.py

```diff
@@ -420,56 +511,46 @@ def plot_state_qsphere(rho, filename=None, show=False):
             ax.plot([0], [0], [0], markerfacecolor=(.5, .5, .5),
                     markeredgecolor=(.5, .5, .5), marker='o', markersize=10,
                     alpha=1)
-            if filename:
-                plt.savefig(filename)
-            elif show:
-                plt.show()
             we[prob_location] = 0
         else:
             break
-    return plt.gcf()
+    plt.tight_layout()
+    plt.close(fig)
+    return fig
 
 
-def plot_state(quantum_state, method='city', filename=None, show=False):
+def plot_state(quantum_state, method='city', figsize=None):
     """Plot the quantum state.
 
     Args:
         quantum_state (ndarray): statevector or density matrix
                                  representation of a quantum state.
         method (str): Plotting method to use.
-        filename (str): the output file to save the plot as. If specified it
-            will save and exit and not open up the plot in a new window. If
-            `bloch` method is used a `-n` will be added to the filename before
-            the extension for each qubit.
-        show (bool): If set to true the rendered image will open in a new
-            window
+        figsize (tuple): Figure size in inches,
+
     Returns:
          matplotlib.Figure: The matplotlib.Figure of the visualization
     Raises:
+        ImportError: Requires matplotlib.
         VisualizationError: if the input is not a statevector or density
         matrix, or if the state is not an multi-qubit quantum state.
     """
-
+    if not HAS_MATPLOTLIB:
+        raise ImportError('Must have Matplotlib installed.')
+    warnings.warn("'plot_state' will be depreciated in terra-0.9.  \
+                  Call the desired plotting routine directly.",
+                  DeprecationWarning)
     # Check if input is a statevector, and convert to density matrix
-    rho = np.array(quantum_state)
-    if rho.ndim == 1:
-        rho = np.outer(rho, np.conj(rho))
-    # Check the shape of the input is a square matrix
-    shape = np.shape(rho)
-    if len(shape) != 2 or shape[0] != shape[1]:
-        raise VisualizationError("Input is not a valid quantum state.")
-    # Check state is an n-qubit state
-    num = int(np.log2(len(rho)))
-    if 2 ** num != len(rho):
-        raise VisualizationError("Input is not a multi-qubit quantum state.")
+    rho = _validate_input_state(quantum_state)
     fig = None
     if method == 'city':
-        fig = plot_state_city(rho, filename=filename, show=show)
+        fig = plot_state_city(rho, figsize=figsize)
     elif method == "paulivec":
-        fig = plot_state_paulivec(rho, filename=filename, show=show)
+        fig = plot_state_paulivec(rho, figsize=figsize)
     elif method == "qsphere":
-        fig = plot_state_qsphere(rho, filename=filename, show=show)
+        fig = plot_state_qsphere(rho, figsize=figsize)
     elif method == "bloch":
```

Comment:
Should we add a new function for the multiqubit bloch vector? Right now when we remove plot_state() we'll lose this functionality.